token-lending/program/src/processor.rs

```diff
@@ -1338,8 +1340,49 @@ fn _withdraw_obligation_collateral<'a>(
         return Err(LendingError::InvalidMarketAuthority.into());
     }
 
+    // account for lending market and reserve rate limiter when withdrawing. this is needed to
+    // support max withdraws.
+    let max_outflow_collateral_amount = if account_for_rate_limiter {
+        let max_outflow_usd = lending_market
+            .rate_limiter
+            .clone() // remaining_outflow is a mutable call, but we don't have mutable access here
```

Comment:
yeah this clone is pretty jank. the alternative is to make the remaining_outflow call immutable.